### ISSUE TITLE
How do you train the bottom up model with attributes

### ISSUE THREAD

**Issue by rico93** [facebookresearch/pythia#5]
Hi:

I want to know how you add predicting attributes in bottom up model,because i currently can't find it in repo. Would you mind providing the code for that ?

Best Wishes
Hanshan Zhang

**Comment by endernewton** [comment]
We followed the same as the original bottom up model did: https://github.com/peteanderson80/bottom-up-attention/blob/master/models/vg/ResNet-101/faster_rcnn_end2end_final/train.prototxt. We predict the attribute with both the ground truth object class embedding and the bottom RoI features. Nothing new.

For the actual code to train the model, it is currently mixed with other efforts we current have. It will take a while to clean that up and release. But it is purely based on the Caffe2 detectron and we follow the same as Anderson et al did -- should not be hard to reproduce on your end.

**Comment by endernewton** [comment]
IIRC, I had to fix a bug for predicting the attributes when implementing that (cannot recall what it exactly it is but it is related to the cross entropy loss). But it may work even work without that bug fixed.

**Comment by rico93** [comment]
Hi，

Can u recall the performance of predicting the attributes ? I want to compare it with my implementation

**Comment by endernewton** [comment]
I did not check the prediction of attributes, I did check the detection performance though, which is in the same ballpark of what reported by Anderson.

**Comment by rico93** [comment]
Thanks for your help. I will look at it. Thanks again
